### PR TITLE
update surveydetails is refactored

### PR DESCRIPTION
**Pull request by anissa-agahchen**
# Overview

## Links to jira tickets

- https://quartech.atlassian.net/browse/BHBC-1078

## This PR contains the following changes

- introduction of survey entities for updating survey details and survey proprietor
- refactoring survey details to work with survey entities
- ability to edit survey_proprietor:
* adding a new record
* deleting an existing record
* updating an existing record

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)
- [x] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [x] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [x] I have performed a self-review of my own code

### Tests

- [x] I have updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

**Comment by NickPhura** [comment]
Getting an error in the API `error: relation "permit" does not exist`.  I think it just needs the latest dev code merged in, as the Permit db changes PR went in on Friday I think.

Edit: Hmm, I was testing in Openshift originally, but I don't get this issue locally.  I did delete all of the openshift pods last night before re-building it (because some un-related issue failed the build) but maybe something is still going on, an old build artifact or something.

File: app/src/features/surveys/view/components/SurveyGeneralInformation.tsx [hunk 1]
```diff
@@ -68,43 +69,47 @@ const SurveyGeneralInformation: React.FC<ISurveyGeneralInformationProps> = (prop
...
+        UPDATE_GET_SURVEY_ENTITIES.survey_details
+      ]);
 
       if (!response) {
         showErrorDialog({ open: true });
         return;
       }
 
-      generalInformationResponseData = response;
+      if (response.survey_details) {
```
Comment (NickPhura) [inline app/src/features/surveys/view/components/SurveyGeneralInformation.tsx]:
```Typescript
  const handleDialogEditOpen = async () => {
    let surveyDetailsResponseData;

    try {
      const response = await biohubApi.survey.getSurveyForUpdate(projectForViewData.id, survey_details?.id, [
        UPDATE_GET_SURVEY_ENTITIES.survey_details
      ]);

      if (!response?.survey_details) {
        showErrorDialog({ open: true });
        return;
      }

      surveyDetailsResponseData = response.survey_details;
    } catch (error) {
      const apiError = error as APIError;
      showErrorDialog({ dialogText: apiError.message, open: true });
      return;
    }

    setSurveyDataForUpdate(surveyDetailsResponseData);
    setGeneralInformationFormData({
      ...surveyDetailsResponseData,
      start_date: getFormattedDate(DATE_FORMAT.ShortDateFormat, surveyDetailsResponseData.start_date),
      end_date: getFormattedDate(DATE_FORMAT.ShortDateFormat, surveyDetailsResponseData.end_date)
    });
    setOpenEditDialog(true);
  };
```

File: api/src/models/survey-update.ts [hunk 2]
```diff
@@ -23,18 +23,52 @@ export class PutSurveyData {
...
+    this.lead_first_name = obj?.survey_details?.biologist_first_name || null;
+    this.lead_last_name = obj?.survey_details?.biologist_last_name || null;
+    this.location_name = obj?.survey_details?.survey_area_name || null;
+    this.geometry = obj?.survey_details?.geometry || null;
+    this.revision_count = obj?.survey_details?.revision_count ?? null;
+  }
+}
+
+/**
+ * Pre-processes PUT /project/{projectId}/survey/{surveyId} survey data for update
```
Comment (sdevalapurkar) [inline api/src/models/survey-update.ts]:
```suggestion
 * Pre-processes PUT /project/{projectId}/survey/{surveyId} survey proprietor data for update
```

File: api/src/models/survey-update.ts [hunk 3]
```diff
@@ -23,18 +23,52 @@ export class PutSurveyData {
...
+ *
+ * @export
+ * @class PutSurveyDetailsData
```
Comment (sdevalapurkar) [inline api/src/models/survey-update.ts]:
```suggestion
 * @class PutSurveyProprietorData
```

File: api/src/models/survey-view-update.ts [hunk 4]
```diff
@@ -55,3 +57,41 @@ export class GetSurveyData {
     this.revision_count = surveyDataItem?.revision_count ?? null;
   }
 }
+
+/**
+ * Pre-processes GET /project/{projectId}/survey/{surveyId} survey proprietor data
+ *
+ * @export
+ * @class GetSurveyDetailsData
```
Comment (sdevalapurkar) [inline api/src/models/survey-view-update.ts]:
```suggestion
 * @class GetSurveyProprietorData
```

File: api/src/paths/project/{projectId}/survey/{surveyId}/view.test.ts [hunk 5]
```diff
@@ -79,36 +83,42 @@ describe('getSurveyForView', () => {
...
+  //   try {
+  //     const result = view.getSurveyForView();
+
+  //     await result(sampleReq, (null as unknown) as any, (null as unknown) as any);
+  //     expect.fail();
+  //   } catch (actualError) {
+  //     expect(actualError.status).to.equal(400);
+  //     expect(actualError.message).to.equal('Failed to build SQL get statement');
+  //   }
+  // });
```
Comment (sdevalapurkar) [inline api/src/paths/project/{projectId}/survey/{surveyId}/view.test.ts]: remove commented code

**Review by sdevalapurkar — APPROVED** [review]
lgtm

**Pull request merged**